homeassistant/components/launch_library/sensor.py

```diff
@@ -39,39 +43,69 @@ async def async_setup_platform(
     async_add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Create the launch sensor."""
-    name = config[CONF_NAME]
-    session = async_get_clientsession(hass)
-    launches = PyLaunches(session)
+    """Import Launch Library configuration from yaml."""
+    _LOGGER.warning(
+        "Loading Launch Library via platform setup is deprecated. Please remove it from your yaml configuration"
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+
+    name = entry.data.get(CONF_NAME, DEFAULT_NAME)
+
+    coordinator = hass.data[DOMAIN]
 
-    async_add_entities([LaunchLibrarySensor(launches, name)], True)
+    async_add_entities(
+        [
+            NextLaunchSensor(coordinator, entry.entry_id, name),
+        ]
+    )
 
 
-class LaunchLibrarySensor(SensorEntity):
-    """Representation of a launch_library Sensor."""
+class NextLaunchSensor(CoordinatorEntity, SensorEntity):
+    """Representation of the next launch sensor."""
 
-    _attr_icon = "mdi:rocket"
+    _attr_attribution = ATTRIBUTION
+    _attr_icon = "mdi:rocket-launch"
 
-    def __init__(self, api: PyLaunches, name: str) -> None:
-        """Initialize the sensor."""
-        self.api = api
+    def __init__(
+        self, coordinator: DataUpdateCoordinator, entry_id: str, name: str
+    ) -> None:
+        """Initialize a Launch Library entity."""
+        super().__init__(coordinator)
         self._attr_name = name
+        self._attr_unique_id = f"{entry_id}_next_launch"
+
+    def get_next_launch(self) -> Launch | None:
+        """Return next launch."""
+        return next((launch for launch in self.coordinator.data), None)
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the state of the sensor."""
+        next_launch = self.get_next_launch()
+        return next_launch.name if next_launch else None
 
-    async def async_update(self) -> None:
-        """Get the latest data."""
-        try:
-            launches = await self.api.upcoming_launches()
-        except PyLaunchesException as exception:
-            _LOGGER.error("Error getting data, %s", exception)
-            self._attr_available = False
-        else:
-            if next_launch := next((launch for launch in launches), None):
-                self._attr_available = True
```

Comment:
Add a available property that returns false if `.get_next_launch` returns `None`